intro_to_ada.rst

```diff
@@ -1904,23 +1904,646 @@ Abstract data types
 Generics
 ========
 
-Generic declaration
--------------------
-
-Generic body
+Introduction
 ------------
 
+Generics are used for metaprogramming in Ada. They are useful for abstract
+algorithms that share common properties.
+
+Generics can be used for subprograms or packages. A generic is declared
+by using the keyword ``generic``. For example:
+
+.. code-block:: ada
+
+    procedure Show_Simple_Generic is
+
+       generic
+          type T is private;
+          --  Declaration of formal types and objects
+       procedure Operator (X : in out T);
+       --  This could be one of the following:
+       --  <procedure | function | package>
+
+       procedure Operator (X : in out T) is null;
+
+    begin
+       null;
+    end Show_Simple_Generic;
+
+Formal type declaration
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Formal types are abstractions of a specific type. We may, for example,
+want to create an algorithm that works on any integer type, or even on
+any type at all, independently of being a numeric type or not. The
+following example declares a formal type ``T`` for the ``Set`` procedure.
+
+.. code-block:: ada
+
+    procedure Show_Formal_Type_Declaration is
```

Comment:
For this example and all the others, I think you can do without the enclosing procedure, just put them at the top level, with file names in comments, and we'll figure out how to make those runnable

```ada
--  set.ads
generic
   type T is private;
   --  T is a formal type that indicates that any type can be used,
   --  be it a numeric type or, for example, a record.
procedure Set (E : in T);

--  set.adb
procedure Set (E : in T) is null;
```